rust-src/concordium_base/src/hashes.rs

```diff
@@ -23,6 +23,16 @@ pub struct HashBytes<Purpose> {
     _phantom:         PhantomData<Purpose>,
 }
 
+impl<Purpose> PartialEq for HashBytes<Purpose> {
```

Comment:
I think you added this after the initial review. I am not sure what the purpose is. All Marker types we have implement PartialEq and Eq, so you will get this for those.

In any case, if this is a manual implementation then also `Eq` should be. Since the `Eq` instance now is `<Purpose: Eq> Eq<HashBytes<Purpose>>`.

Reply:
I added an explicit implementation for `Eq` as well now.

Reply:
@abizjak , assuming all the checks complete successfully, could you confirm that things are OK now?

Reply:
See new comment. THe PartialEq instance should be fixed. Otherwise good.

Reply:
On another note, an issue has been filed regarding the `derive_hash_xor_eq` clippy check: https://github.com/rust-lang/rust-clippy/issues/2627

Apparently it doesn't make much sense to deny deriving `PartialEq` with an explicit definition of `Hash`, but it's only just been closed recently.

Reply:
Ok.